### ISSUE TITLE
API methods open() not firing in events input focus() 

### ISSUE THREAD

**Issue by micabal** [TarekRaafat/autoComplete.js#233]
Version: v10.0.4

When using the [example](https://tarekraafat.github.io/autoComplete.js/demo/) to fire open() or start() on focus, nothing happens.

_The following does **NOT** work:_
```
events: {
        input: {
            focus: () => {
          
                if (autoCompleteJS.input.value.length) autoCompleteJS.open();

            },
        },
    },
```

**_But this works fine:_**
```
events: {
    input: {
        focus: () => {
         
            const ctx = autoCompleteJS;
            const eventEmitter = (function (name, ctx) {
                ctx.input.dispatchEvent(new CustomEvent(name, {
                  bubbles: true,
                  detail: ctx.feedback,
                  cancelable: true
                }));
              });

            if (ctx.input.value.length) {

                ctx.input.parentNode
                .querySelector("ul").hidden = false;


                if (ctx.isOpen) return;
                ctx.wrapper.setAttribute("aria-expanded", true);
                ctx.list.removeAttribute("hidden");
                ctx.isOpen = true;
                eventEmitter("start", ctx);

            }

        },
    },
},

```

Let me know if you need further details.
PS: Thanks for the amazing job!

**Comment by TarekRaafat** [comment]
Hello @micabal,

Thanks a lot for reporting this issue! there was another issue somehow close to your issue that was reported today #230

But unfortunately, I'm trying to reproduce your issue with no luck. I keep getting the first code snippet working for me, and the live [example](https://tarekraafat.github.io/autoComplete.js/demo/) also works fine.

Could you please share more details about the used browser, any errors you get, etc.?

Cheers, and have a nice day! :)

**Comment by micabal** [comment]
Thanks for your prompt reply. 
Browser is Chrome (But I do not think it is relevant as the example works fine 😄 )


I have the following error (sorry I though it was not relevant but I may be wrong). 
GET http://localhost:35729/livereload.js?snipver=1 net::ERR_CONNECTION_REFUSED


Also, I have in all the arguments of the`__proto__` methods of the `console.log(autoCompleteJS);`, the following `Exception: TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode`
The references of the prototypes are [here](https://github.com/TarekRaafat/autoComplete.js/blob/f7ada6cb0d1419278bc94e0071a0bf2ea96dfcde/docs/demo/js/autoComplete.js#L641)



Finally, I attach below the `console.log(autoCompleteJS);`. Let me know if it is helpful or if you need anything else.


```
{
    "options": {
        "searchEngine": "loose",
        "debounce": 300,
        "placeHolder": "Restaurant, Vegan, Vintage...",
        "data": {
            "keys": [
                "srcExp"
            ]
        },
        "resultsList": {
            "maxResults": 12,
            "noResults": true
        },
        "resultItem": {
            "highlight": {
                "render": true
            }
        },
        "events": {
            "input": {}
        }
    },
    "id": 1,
    "name": "autoComplete",
    "threshold": 1,
    "debounce": 300,
    "resultsList": {
        "position": "afterend",
        "tag": "ul",
        "maxResults": 12,
        "noResults": true,
        "id": "autoComplete_list_1"
    },
    "resultItem": {
        "tag": "li",
        "highlight": {
            "render": true
        },
        "id": "autoComplete_result"
    },
    "searchEngine": "loose",
    "placeHolder": "Restaurant, Vegan, Vintage...",
    "data": {
        "keys": [
            "srcExp"
        ],
        "store": [
            {
                "id": 0,
                "srcExp": "Restaurant",
                "ranking": 1,
                "count": 1,
                "exper_image_name": null,
                "exper_type": 0,
                "city": null
            },
            {
                "id": 0,
                "srcExp": "Restaurant and private beach",
                "ranking": 1,
                "count": 1,
                "exper_image_name": null,
                "exper_type": 0,
                "city": null
            },
            {
                "id": 0,
                "srcExp": "Restoration",
                "ranking": 1,
                "count": 1,
                "exper_image_name": null,
                "exper_type": 0,
                "city": null
            }
        ]
    },
    "events": {
        "input": {},
        "list": {}
    },
    "input": {
        "jQuery3410234448499936887432": {}
    },
    "wrapper": {},
    "list": {},
    "feedback": {
        "query": "res",
        "matches": [
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>taurant",
                "value": {
                    "id": 0,
                    "srcExp": "Restaurant",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            },
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>taurant and private beach",
                "value": {
                    "id": 0,
                    "srcExp": "Restaurant and private beach",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            },
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>toration",
                "value": {
                    "id": 0,
                    "srcExp": "Restoration",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            }
        ],
        "results": [
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>taurant",
                "value": {
                    "id": 0,
                    "srcExp": "Restaurant",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            },
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>taurant and private beach",
                "value": {
                    "id": 0,
                    "srcExp": "Restaurant and private beach",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            },
            {
                "match": "<mark>R</mark><mark>e</mark><mark>s</mark>toration",
                "value": {
                    "id": 0,
                    "srcExp": "Restoration",
                    "ranking": 1,
                    "count": 1,
                    "exper_image_name": null,
                    "exper_type": 0,
                    "city": null
                },
                "key": "srcExp"
            }
        ]
    },
    "cursor": -1,
    "isOpen": false
}
```

**Comment by TarekRaafat** [comment]
My bad, the `livereload` was not supposed to be included in the release it's just for development.

The issue is resolved in [v10.1.0](https://github.com/TarekRaafat/autoComplete.js/releases/tag/v10.1.0), please check it and let me know how it goes.

**Comment by micabal** [comment]
Everything works fine on [v10.1.1](https://github.com/TarekRaafat/autoComplete.js/releases/tag/v10.1.1).

Thanks !.. You do an amazing job and follow-up 🚀
